docs/sql-programming-guide.md

```diff
@@ -581,6 +581,15 @@ Configuration of Parquet can be done using the `setConf` method on SQLContext or
   </td>
 </tr>
 <tr>
+  <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
+  <td>true</td>
+  <td>
+    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
+    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
+    flag tells Spark SQL to interpret INT96 data as a timestamp to provide compatibility with these systems.
+  </td>
+</tr>
```

Comment:
From my digging only Parquet-format 2.2 has the TIMESTAMP and TIMESTAMP_MILLS types. Cloudera is still on 1.5.0
Hive/Impala has been writing this INT96 nano sec format that's different.

--- Original Message ---

From: "Michael Armbrust" <notifications@github.com>
Sent: January 20, 2015 11:25 AM
To: "apache/spark" <spark@noreply.github.com>
Cc: "Felix Cheung" <felixcheung_m@hotmail.com>
Subject: Re: [spark] [SPARK-4987] [SQL] parquet timestamp type support (#3820)

> @@ -581,6 +581,15 @@ Configuration of Parquet can be done using the `setConf` method on SQLContext or
>    </td>
>  </tr>
>  <tr>
> +  <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
> +  <td>true</td>
> +  <td>
> +    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
> +    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
> +    flag tells Spark SQL to interpret INT96 data as a timestamp to provide compatibility with these systems.
> +  </td>
> +</tr>

Yeah, I agree that it's weird though.  Perhaps we should ask the parquet
list why they don't support the int 96 version.
On Jan 20, 2015 11:21 AM, "Cheng Lian" <notifications@github.com> wrote:

> In docs/sql-programming-guide.md
> <https://github.com/apache/spark/pull/3820#discussion-diff-23247492>:
>
> > @@ -581,6 +581,15 @@ Configuration of Parquet can be done using the `setConf` method on SQLContext or
> >    </td>
> >  </tr>
> >  <tr>
> > +  <td><code>spark.sql.parquet.int96AsTimestamp</code></td>
> > +  <td>true</td>
> > +  <td>
> > +    Some Parquet-producing systems, in particular Impala, store Timestamp into INT96. Spark would also
> > +    store Timestamp as INT96 because we need to avoid precision lost of the nanoseconds field. This
> > +    flag tells Spark SQL to interpret INT96 data as a timestamp to provide compatibility with these systems.
> > +  </td>
> > +</tr>
>
> Oh, I see the difference here. Double checked, Parquet only provides
> TIMESTAMP and TIMESTAMP_MILLIS
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/3820/files#r23247492>.
>

---
Reply to this email directly or view it on GitHub:
https://github.com/apache/spark/pull/3820/files#r23247815